<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SubtitleExtraction</class>
 <widget class="QWidget" name="SubtitleExtraction">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>880</width>
    <height>620</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>SubtitleExtraction</string>
  </property>
  <layout class="QVBoxLayout" name="rootLayout">
   <property name="spacing">
    <number>16</number>
   </property>
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <widget class="QFrame" name="cardFrame">
     <layout class="QVBoxLayout" name="cardLayout">
      <property name="spacing">
       <number>12</number>
      </property>
      <property name="leftMargin">
       <number>16</number>
      </property>
      <property name="topMargin">
       <number>16</number>
      </property>
      <property name="rightMargin">
       <number>16</number>
      </property>
      <property name="bottomMargin">
       <number>16</number>
      </property>
      <item>
       <layout class="QHBoxLayout" name="titleRowLayout">
        <item>
         <widget class="QLabel" name="titleLabel">
          <property name="styleSheet">
           <string notr="true">font: 12pt &quot;Microsoft YaHei&quot;; font-weight: 600;</string>
          </property>
          <property name="text">
           <string>语音识别与字幕提取</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="titleSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QToolButton" name="toolsCheckButton">
          <property name="minimumSize">
           <size>
            <width>28</width>
            <height>28</height>
           </size>
          </property>
          <property name="toolTip">
             <string>检查并更新依赖（yt-dlp / whisper / ffmpeg）</string>
          </property>
          <property name="icon">
           <iconset resource="../../../resources/style.qrc">
            <normaloff>:/new/prefix1/style/QSS_IMG/object-select-symbolic.symbolic.png</normaloff>:/new/prefix1/style/QSS_IMG/object-select-symbolic.symbolic.png</iconset>
          </property>
          <property name="iconSize">
           <size>
            <width>18</width>
            <height>18</height>
           </size>
          </property>
          <property name="autoRaise">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QGridLayout" name="modelLanguageLayout">
        <property name="horizontalSpacing">
         <number>12</number>
        </property>
        <property name="verticalSpacing">
         <number>10</number>
        </property>
        <item row="0" column="0">
         <widget class="QLabel" name="modelLabel">
          <property name="styleSheet">
           <string notr="true">color: #6E737A;</string>
          </property>
          <property name="text">
           <string>语音识别模型</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QComboBox" name="modelComboBox">
          <property name="minimumSize">
           <size>
            <width>320</width>
            <height>0</height>
           </size>
          </property>
          <item>
           <property name="text">
            <string>Whisper Tiny</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Whisper Base</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Whisper Small</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Whisper Medium</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Whisper Large V2</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Whisper Large V3</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QPushButton" name="importModelButton">
          <property name="text">
           <string>打开模型目录</string>
          </property>
         </widget>
        </item>
        <item row="0" column="3">
         <widget class="QCheckBox" name="gpuCheckBox">
          <property name="text">
           <string>GPU</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="languageLabel">
          <property name="styleSheet">
           <string notr="true">color: #6E737A;</string>
          </property>
          <property name="text">
           <string>语言</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QComboBox" name="languageComboBox">
          <item>
           <property name="text">
            <string>自动检测</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>中文</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>English</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>日本語</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>한국어</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Español</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Français</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Deutsch</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Русский</string>
           </property>
          </item>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QGroupBox" name="transcribeGroupBox">
        <property name="title">
           <string>转写任务</string>
        </property>
        <layout class="QVBoxLayout" name="transcribeLayout">
         <property name="spacing">
          <number>10</number>
         </property>
         <item>
          <layout class="QHBoxLayout" name="inputRowLayout">
           <item>
            <widget class="QLineEdit" name="inputLineEdit">
             <property name="placeholderText">
              <string>选择音视频文件</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="inputBrowseButton">
             <property name="text">
              <string>...</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="formatRowLayout">
           <item>
            <widget class="QLabel" name="formatLabel">
             <property name="styleSheet">
              <string notr="true">color: #6E737A;</string>
             </property>
             <property name="text">
              <string>中间文件目录</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="tempDirLineEdit">
             <property name="minimumSize">
              <size>
               <width>220</width>
               <height>0</height>
              </size>
             </property>
             <property name="placeholderText">
              <string>分段音频与中间字幕暂存目录</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="tempDirBrowseButton">
             <property name="text">
              <string>...</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="formatSpacer">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="outputRowLayout">
             <item>
              <widget class="QLabel" name="outputFormatLabel">
               <property name="styleSheet">
                <string notr="true">color: #6E737A;</string>
               </property>
               <property name="text">
                <string>输出格式</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="outputFormatComboBox">
               <item>
                <property name="text">
                 <string>TXT</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>TXT（带时间）</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>SRT</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>WebVTT</string>
                </property>
               </item>
               <property name="currentIndex">
                <number>2</number>
               </property>
              </widget>
             </item>
           <item>
            <widget class="QLineEdit" name="outputLineEdit">
             <property name="placeholderText">
              <string>最终字幕输出目录</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="outputBrowseButton">
             <property name="text">
              <string>...</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="actionRowLayout">
           <item>
            <spacer name="actionSpacer">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QCheckBox" name="debugConsoleCheckBox">
             <property name="text">
              <string>完成后清理中间文件</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="transcribeButton">
             <property name="minimumSize">
              <size>
               <width>120</width>
               <height>32</height>
              </size>
             </property>
             <property name="text">
              <string>开始转写</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QFrame" name="logFrame">
     <property name="styleSheet">
      <string notr="true">background-color: #FFFFFF; border-radius: 10px;</string>
     </property>
     <layout class="QVBoxLayout" name="logLayout">
      <property name="spacing">
       <number>8</number>
      </property>
      <property name="leftMargin">
       <number>16</number>
      </property>
      <property name="topMargin">
       <number>16</number>
      </property>
      <property name="rightMargin">
       <number>16</number>
      </property>
      <property name="bottomMargin">
       <number>16</number>
      </property>
      <item>
       <widget class="QLabel" name="logTitleLabel">
        <property name="styleSheet">
         <string notr="true">font: 11pt &quot;Microsoft YaHei&quot;; font-weight: 600;</string>
        </property>
        <property name="text">
         <string>任务日志与进度</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QTextEdit" name="logTextEdit">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>140</height>
         </size>
        </property>
        <property name="placeholderText">
         <string/>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="bottomButtonLayout">
     <item>
      <widget class="QPushButton" name="backButton">
       <property name="text">
        <string>返回</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="audioCaptureButton">
       <property name="text">
        <string>音频采集</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="bottomSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="closeButton">
       <property name="text">
        <string>转到下一步</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="../../../resources/style.qrc"/>
 </resources>
 <connections/>
</ui>
